routing/rt/rt.go

```diff
@@ -0,0 +1,324 @@
+	for _, op := range *diffBuff {
+		if op.Operation == ADD_CHANNEL {
+			rt.AddChannel(op.Id1, op.Id2, op.EdgeID, op.Info)
+		} else if op.Operation == REMOVE_CHANNEL {
+			rt.RemoveChannel(op.Id1, op.Id2, op.EdgeID)
+		}
+	}
+}
+
+func (rt *RoutingTable) Nodes() []graph.ID {
+	return rt.G.GetIDs()
+}
+
+func (rt *RoutingTable) NumberOfNodes() int {
+	return rt.G.GetIDCount()
+}
+
+func (rt *RoutingTable) AllChannels() []Channel {
+	edges := rt.G.GetUndirectedEdges()
+	channels := make([]Channel, 0, len(edges))
+	// fmt.Println(len(edges))
+	for _, edge := range edges {
+		info := edge.Info()
+		channels = append(channels, Channel{
+			Id1:    edge.Source(),
+			Id2:    edge.Target(),
+			EdgeID: edge.ID(),
+			Info:   info,
+		})
+	}
+	return channels
+}
+
+func (rt *RoutingTable) AddTable(rt1 *RoutingTable) {
+	newChannels := rt1.AllChannels()
+	for _, channel := range newChannels {
+		rt.AddChannel(channel.Id1, channel.Id2, channel.EdgeID, channel.Info.Copy())
+	}
+}
+
+func (rt *RoutingTable) Copy() *RoutingTable {
+	// TODO: add tests
+	channels := rt.AllChannels()
+	newRT := NewRoutingTable()
+	newRT.AddNode(rt.Nodes()...)
+	for _, channel := range channels {
+		newRT.AddChannel(channel.Id1, channel.Id2, channel.EdgeID, channel.Info.Copy())
+	}
+	return newRT
+}
+
+func (rt *RoutingTable) AddTableFiltered(rt2 *RoutingTable, Node graph.ID, MaxDist int) {
+	// newNodes := rt2.Nodes()
+	rtCopy := rt.Copy()
+	rtCopy.AddTable(rt2)
+	distances, _, err := rtCopy.G.Bfs(Node)
+	if err != nil {
+		panic(err)
+	}
+	IsClose := func(u, v graph.ID) bool {
+		distu, ok1 := distances[u]
+		distv, ok2 := distances[v]
+		if !ok1 || !ok2 {
+			return false
+		}
+		return (distu + distv) <= 2*MaxDist
+	}
+	newChannels := rt2.AllChannels()
+	for _, channel := range newChannels {
+		if !rt.HasChannel(channel.Id1, channel.Id2, channel.EdgeID) && IsClose(channel.Id1, channel.Id2) {
+			rt.AddChannel(channel.Id1, channel.Id2, channel.EdgeID, channel.Info)
+		}
+	}
+}
+
+func (rt *RoutingTable) ApplyDiffBuffFiltered(diffBuff *DifferenceBuffer, Node graph.ID, MaxDist int) {
+	rtCopy := rt.Copy()
+	rtCopy.ApplyDiffBuff(diffBuff)
+	distances, _, err := rtCopy.G.Bfs(Node)
+	if err != nil {
+		panic(err)
+	}
+	IsClose := func(u, v graph.ID) bool {
+		distu, ok1 := distances[u]
+		distv, ok2 := distances[v]
+		if !ok1 || !ok2 {
+			return false
+		}
+		return (distu + distv) <= 2*MaxDist
+	}
+	filteredDiffBuff := NewDifferenceBuffer()
+	for _, chOp := range *diffBuff {
+		if chOp.Operation == REMOVE_CHANNEL || (chOp.Operation == ADD_CHANNEL && IsClose(chOp.Id1, chOp.Id2)) {
+			*filteredDiffBuff = append(*filteredDiffBuff, chOp)
+		}
+	}
+	rt.ApplyDiffBuff(filteredDiffBuff)
+}
+
+func AddressDistance(x1, x2 graph.ID) int {
+	// TODO: INSERT CORRECT FUNCTION. It is important only for beacon-related functionality.
+	return 1
+}
+
+func (rt *RoutingTable) ClosestAddressNode(dest graph.ID, exclude map[graph.ID]struct{}) *graph.ID {
```

Comment:
Since the beacon functionality has been nixed from the initial scheme, this function is no longer necessary.